js/navbar/tabColor.js

```diff
@@ -1,149 +1,123 @@
+      largestPixelSet = k
+      ct = pixels[k]
+    }
+  }
 
-var hours = new Date().getHours() + (new Date().getMinutes() / 60)
+  var res = largestPixelSet.split(',')
 
-// we cache the hours so we don't have to query every time we change the color
+  for (var i = 0; i < res.length; i++) {
+    res[i] = parseInt(res[i])
+  }
 
-setInterval(function () {
-  var d = new Date()
-  hours = d.getHours() + (d.getMinutes() / 60)
-}, 4 * 60 * 1000)
-
-function updateTabColor (favicons, tabId) {
-  // private tabs always use a special color, we don't need to get the icon
-  if (tabs.get(tabId).private === true) {
-    return
+  // dim the colors late at night or early in the morning, or when dark mode is enabled
+  var colorChange = 1
+  if (hours > 20) {
+    colorChange -= 0.015 * Math.pow(2.75, hours - 20)
+  } else if (hours < 6.5) {
+    colorChange -= -0.15 * Math.pow(1.36, hours) + 1.15
   }
-  requestIdleCallback(function () {
-    getColor(favicons[0], function (c) {
-      // dim the colors late at night or early in the morning, or when dark mode is enabled
-      var colorChange = 1
-      if (hours > 20) {
-        colorChange -= 0.015 * Math.pow(2.75, hours - 20)
-      } else if (hours < 6.5) {
-        colorChange -= -0.15 * Math.pow(1.36, hours) + 1.15
-      }
 
-      if (window.isDarkMode) {
-        colorChange = Math.min(colorChange, 0.58)
-      }
+  if (window.isDarkMode) {
+    colorChange = Math.min(colorChange, 0.58)
+  }
 
-      c[0] = Math.round(c[0] * colorChange)
-      c[1] = Math.round(c[1] * colorChange)
-      c[2] = Math.round(c[2] * colorChange)
+  res[0] = Math.round(res[0] * colorChange)
+  res[1] = Math.round(res[1] * colorChange)
+  res[2] = Math.round(res[2] * colorChange)
 
-      var cr = 'rgb(' + c[0] + ',' + c[1] + ',' + c[2] + ')'
+  return res
+}
 
-      var obj = {
-        r: c[0] / 255,
-        g: c[1] / 255,
-        b: c[2] / 255
-      }
+function getHours () {
+  return new Date().getHours() + (new Date().getMinutes() / 60)
+}
 
-      var textclr = getTextColor(obj)
+var hours = getHours()
 
-      tabs.update(tabId, {
-        backgroundColor: cr,
-        foregroundColor: textclr
-      })
+// we cache the hours so we don't have to query every time we change the color
+setInterval(function () {
+  hours = getHours()
+}, 5 * 60 * 1000)
 
-      if (tabId === tabs.getSelected()) {
-        updateColorPalette()
-      }
-      return
-    })
-  }, {
-    timeout: 1000
-  })
+function getRGBString (c) {
+  return 'rgb(' + c[0] + ',' + c[1] + ',' + c[2] + ')'
 }
 
-// generated using http://harthur.github.io/brain/
 var getTextColor = function (bgColor) {
-  var output = runNetwork(bgColor)
+  var obj = {
+    r: bgColor[0] / 255,
+    g: bgColor[1] / 255,
+    b: bgColor[2] / 255
+  }
+  var output = runNetwork(obj)
   if (output.black > 0.5) {
     return 'black'
   }
   return 'white'
 }
 
+// generated using http://harthur.github.io/brain/
 var runNetwork = function anonymous (input) {
```

Comment:
Might be a copy-paste typo, but looks out of place here - `var runNetwork = function anonymous (input)`. I also feel it should get out of the `tabColor.js` file, it occupies a good 60 lines here and is not strictly bound to the tab color logic, it's rather an implementation detail.

Reply:
This is actually external code, so I've moved it to ext/tabColor/tabColor.js